src/idls/thrift/src/t_yarp_generator.cc

```diff
@@ -1840,6 +1842,22 @@ void t_yarp_generator::generate_struct(t_struct* tstruct)
         yarp_api_keyword = annotations.at("yarp.api.keyword");
     }
 
+    std::string yarp_type_name{};
+    if (annotations.find("yarp.type.name") != annotations.end()) {
+        yarp_type_name = annotations.at("yarp.type.name");
+    }
+    else {
+        yarp_type_name = "yarp/"+name;
+    }
+
+    std::string yarp_type_version{};
+    if (annotations.find("yarp.type.version") != annotations.end()) {
+        yarp_type_version = annotations.at("yarp.type.version");
+    }
+    else {
+        yarp_type_version = "1.0";

```

Comment:
Same about the default version, 1.0 is not a good value, since an unstable type with no annotation defined will have a stable version. I'd stick with enabling this part only if explicitly added by the user.

Reply:
0.0? or empty string?

Reply:
No, I'd just leave it empty, and add the relative calls when generating the code only if it is defined